ape_solidity/compiler.py

```diff
@@ -255,20 +324,23 @@ def _get_compiler_arguments(self, version_map: Dict, base_path: Path) -> Dict[Ve
                     for i, relative_path in import_remappings.items()
                 }
             settings_map[solc_version] = settings
+
         return settings_map
 
     def compile(
         self, contract_filepaths: List[Path], base_path: Optional[Path] = None
     ) -> List[ContractType]:
         contracts_path = base_path or self.config_manager.contracts_folder
         files_by_solc_version = self.get_version_map(contract_filepaths, base_path=contracts_path)
-        settings_map = self._get_compiler_arguments(files_by_solc_version, base_path=contracts_path)
+        compiler_arguments = self._get_compiler_arguments(
+            files_by_solc_version, base_path=contracts_path
+        )
         contract_types: List[ContractType] = []
         solc_versions_by_contract_name: Dict[str, Version] = {}
-        for solc_version, settings in settings_map.items():
+        for solc_version, arguments in compiler_arguments.items():
             files = files_by_solc_version[solc_version]
             logger.debug(f"Compiling using Solidity compiler '{solc_version}'")
-            output = solcx.compile_files([f for f in files], **settings)
+            output = solcx.compile_files([f for f in files], **arguments)
```

Comment:
Similar question here, files is Set[Path], can you pass as a set here?

Reply:
this got done!